plugins/statistics.koplugin/main.lua

```diff
@@ -234,28 +266,138 @@ function getDatesForBookOldFormat(book)
     return generateReadBooksTable(book.title, dates)
 end
 
-function getDatesForBook(book)
+-- Use from here:
+-- http://stackoverflow.com/questions/1340230/check-if-directory-exists-in-lua/21637668#21637668
+function isDir(name)
```

Comment:
You should be able to just check the return value of `lfs.attributes(name)`. If the directory does not exist, it will return `nil`.